test/alternator/run.py

```diff
@@ -4,8 +4,8 @@
 import os
 import requests
 import time
-import cassandra.cluster
-import cassandra.auth
+import cassandra.cluster # type: ignore
+import cassandra.auth # type: ignore
```

Comment:
Why would it be a mypy bug? Mypy correctly reports that it cannot find stubs with type definitions for the `cassandra` project, which is indeed the case - Cassandra driver unfortunately does not provide mypy stubs.

The way to silence this error is to add `# type: ignore` on the appropriate line.
It's the standard way to silence errors reported by Python static code analysis tools, for example you can find pieces like `# pylint: disable=broad-except` in our codebase.

> I hope you don't take it as a personal offense (and instead, that the mypy authors take it upon themselves) - this patch is ugly, makes the code uglier, and it is 100% certain that someone will introduce new code in the future that forgets to add these comments and the whole mypy house of cards crumbles.

No, it won't crumble. Mypy will just report an additional error on this newly added import, and if someone's annoyed by the error, they can silence it, but it doesn't affect other things.

I'll look for a way to do a project-wide `# type: ignore` for the cassandra imports, maybe there's a way to do this through a config file in the project root or sth.

Reply:
> Why would it be a mypy bug? Mypy correctly reports that it cannot find stubs with type definitions for the `cassandra` project, which is indeed the case - Cassandra driver unfortunately does not provide mypy stubs.

Because mypy is a debugging tool. Maybe it's fashionable, and maybe you like it, but it's just a tool. You shouldn't have to write your code to fit debugging tool - the debugging tool should fit actual code.

> The way to silence this error is to add `# type: ignore` on the appropriate line. It's the standard way to silence errors reported by Python static code analysis tools, for example you can find pieces like `# pylint: disable=broad-except` in our codebase.

There is a difference between a comment that says "this code really looks suspicious, but I promise it's fine", and a comment that says basically  "I know this library did not buy into mypy, so I withdraw all claims from mypy to debug it". This sort of stuff should have been in some mypy configuration file, not in my code. I see below that you are looking for a way to have a separate configuration file with those configurations (e.g., which libraries do not have mypy support), and if you can find one, I think it will be better.

In general, it's not pretty when one developers choose one tool (mypy) another uses another tool (pylint) and a third uses another tool (autopep8), and then all of them start sending reformatting patches and adding ugly comments to the code so their favorite tool will be happy. I consider all of this, nonsense. Our code also doesn't have emacs options, Eclipse configurations, and so on, inside the code, and people don't send patches to change the code to look the way Eclipse of CLion or whatever likes it.

But all that being said, I admit that while I find these comments ugly and unnecessary, they are not a disaster, and can be accepted if there are not a huge number of them.


> I'll look for a way to do a project-wide `# type: ignore` for the cassandra imports, maybe there's a way to do this through a config file in the project root or sth.

Good idea (hoping it is possible).